src/components/editor/property/ResourceProperty.jsx

```diff
@@ -56,8 +63,8 @@ export class ResourceProperty extends Component {
 
         propertyReduxPath.push(rtProperty.propertyURI)
         const payload = { reduxPath: propertyReduxPath, property: rtProperty }
+        this.dispatchPayloads.push(payload)
```

Comment:
This pushes onto the array, but "componentDidMount" has already been called, so this won't do anything, right?

Reply:
- the tests failed before I added the calls to `initNewResourceTemplate` in componentDidMount();  there are tests expecting it to be called ...  but maybe they aren't actually testing the redux state, just that the method is called.

Does this imply I need to add a test to show the redux store is correct after render is called?

(I'm increasingly concerned our test coverage data is misleading - that we have code lines being executed by our tests, but the tests aren't sufficient to show the app is working properly ...)

Reply:
@jcoyne "The componentDidMount() method runs after the component output has been rendered to the DOM."  https://reactjs.org/docs/state-and-lifecycle.html

Reply:
Thanks for the clarification. 👍

Reply:
I think this PR is incorrect *IFF* we ever re-render a ResourceProperty, but I don't believe we do???

Reply:
If this re-pushes every time we re-render, does `dispatchPayload` grow without bounds?

Reply:
I don't think we ever do re-render.

Reply:
more from the docs https://reactjs.org/docs/react-component.html#componentdidmount

> `componentDidMount()` is invoked immediately after a component is mounted (inserted into the tree). Initialization that requires DOM nodes should go here. If you need to load data from a remote endpoint, this is a good place to instantiate the network request.

based on that, and naomi's documentation find above, that implies to me that `componentDidMount()` runs after the output has been rendered to the DOM, but only for the render that happens after the first time the component mounts.  it doesn't _seem_ to me, based on what i've seen, like that function runs after every render.

i'm, uh, not even entirely sure this comment is helpful... but left it just in case. 🙂

Reply:
agree: componentDidMount() is only run for initial render.  

have no conflicting evidence that we would ever re-render this component.  adding a comment to that effect.